### PR TITLE
Separators: Implement html chunking strategy.

### PR DESCRIPTION
**Pull request by cpursley**

### CODE REVIEW

File: lib/text_chunker.ex [hunk 1]
```diff
@@ -21,20 +21,21 @@ defmodule TextChunker do
...
     :php,
+    :plaintext,
     :python,
-    :vue,
-    :javascript,
-    :typescript
+    :rtf,
+    :ruby,
+    :typescript,
+    :vue
```
Comment (stuartjohnpage) [inline lib/text_chunker.ex:38]: Thank you for organising these!

File: lib/text_chunker/strategies/recursive_chunk/separators.ex [hunk 2]
```diff
@@ -157,14 +137,38 @@ defmodule TextChunker.Strategies.RecursiveChunk.Separators do
...
+      "<h5",
+      "<h6",
+      "<p",
+      "<ul",
+      "<ol",
+      "<li",
+      "<article",
+      "<section",
+      "<table"
+    ] ++
```
Comment (stuartjohnpage) [inline lib/text_chunker/strategies/recursive_chunk/separators.ex:159]: I'm curious, why is it splitting on `<h1` rather than `<h1>`?

File: lib/text_chunker/strategies/recursive_chunk/separators.ex [hunk 3]
```diff
@@ -157,14 +137,38 @@ defmodule TextChunker.Strategies.RecursiveChunk.Separators do
...
+      empty_and_new_line_separators()
   end
 
   def get_separators(:typescript), do: get_separators(:javascript)
 
   def get_separators(format) when format in @plaintext_formats, do: get_separators(:plaintext)
+
+  defp empty_and_new_line_separators do
```
Comment (stuartjohnpage) [inline lib/text_chunker/strategies/recursive_chunk/separators.ex:167]: nice

**Review by stuartjohnpage — APPROVED** [review]
This looks great: one question, but no blockers. Thank you again!

File: lib/text_chunker/strategies/recursive_chunk/separators.ex (hunk 2 above)
Comment (cpursley) [reply lib/text_chunker/strategies/recursive_chunk/separators.ex:159]: I was just being lazy and following the other examples (vue).

Comment (stuartjohnpage) [reply lib/text_chunker/strategies/recursive_chunk/separators.ex:159]: It just struck me while driving that this will also split on any HMTL tags that have attrs! So it makes way more sense than adding in the closing `>`.

**Review by grossvogel — APPROVED** [review]
This is great, thank you!

My only inclination to nit-pick is around whether `<article` and `<section` should be higher-up in the separator list. But the more I think about it, I'm concluding that documents that use those elements according to the spec are most likely _also_ going to be using the `<h...` tags in a way that gives us _better_ information about how to group the content.

tl;dr; LGTM

**Comment by cpursley** [comment]
I actually thought about not including <article> and <section> - perhaps they are unnecessary. Thoughts?

**Comment by grossvogel** [comment]
@cpursley I can't convince myself that moving (or removing) them would be a definite improvement, so I'm happy with this until somebody comes along with a stronger argument 😄 

Thanks so much for this contribution!

**Comment by cpursley** [comment]
You're welcome. Y'all should do a detailed blog post about RAG stuff in Elixir (how you're working with the chunked data, etc)!
